permissions/resource_access_control_rule_set.go

```diff
@@ -0,0 +1,139 @@
+package permissions
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	"github.com/databricks/databricks-sdk-go/apierr"
+	"github.com/databricks/databricks-sdk-go/service/iam"
+	"github.com/databricks/terraform-provider-databricks/common"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceAccessControlRuleSet() *schema.Resource {
+	s := common.StructToSchema(
+		iam.RuleSetUpdateRequest{},
+		func(m map[string]*schema.Schema) map[string]*schema.Schema {
+			m["etag"].Required = false
+			m["etag"].Computed = true
+			m["grant_rules"].Type = schema.TypeSet
+			common.MustSchemaPath(m, "grant_rules", "principals").Type = schema.TypeSet
+
+			return m
+		})
+	readFromWsOrAcc := func(ctx context.Context, c *common.DatabricksClient, getRuleSetReq iam.GetRuleSetRequest) (*iam.RuleSetResponse, error) {
```

Comment:
@tanmay-db Another use case of two implementations based on whether the provider is account-level or workspace-level. It may be useful to expose some utility function like
```go
func ContextSpecificAPI[T any](func (context.Context, *databricks.WorkspaceClient) error, func (context.Context, *databricks.AccountClient) error) T, error
```
which handles checking for account ID, fetching the client and handling errors associated with that step, reducing the code to purely business logic for the resource implementer.